### PR TITLE
Diff color of attributes and processing instructions

### PR DESCRIPTION
**Pull request by AirQuick**
Following #662, this pull request applies a similar convention to attributes and processing instructions.

#### Attributes
|  | current master | this pull request |
|--|--|--|
| exact match | name=value | name=value |
| name match <br/> (different value) | name=value | name=value |
| orphan | name=value | name=value |

#### Processing instructions

|  | current master | this pull request |
|--|--|--|
| exact match | green | &lt;?name value?> |
| name match <br/> (different value) | solid pink | &lt;?name value?> |
| value match <br/> (different name) | solid pink | &lt;?name value?> |
| no match | solid pink | &lt;?name value?>` |

### Example

See the diff under `When the result contains attribute,` and `When the result contains processing instructions,`:

* [current master](https://htmlpreview.github.io/?https://github.com/xspec/xspec/blob/a4f549c3901330ee9948073b47362ea9e489a19b/test/end-to-end/cases/expected/stylesheet/xspec-serialize-result.html#ELEM-652)
* [this pull request](https://htmlpreview.github.io/?https://github.com/xspec/xspec/blob/9d6a33c26356301f597625053bc1b3b4dd6a2435/test/end-to-end/cases/expected/stylesheet/xspec-serialize-result.html#ELEM-666)

### CODE REVIEW

File: src/reporter/format-utils.xsl [hunk 1]
```diff
@@ -119,16 +119,20 @@
   </xsl:for-each>
 
   <!-- Output attributes while performing comparison -->
-  <xsl:for-each select="@*">
+  <xsl:for-each select="attribute()">
+    <xsl:variable name="attribute-to-compare-with" as="attribute()?"
+      select="$node-to-compare-with/attribute()[node-name(.) eq node-name(current())]" />
+
+    <xsl:variable name="display-value" as="xs:string"
+      select="replace(replace(., '&quot;', '&amp;quot;'), '\s(\s+)', '&#x0A;$1')" />
```
Comment (galtm) [inline src/reporter/format-utils.xsl]: Did you consider including the surrounding quotation marks in `display-value` (and removing them from the `xsl:text` instructions on lines 143 and 154)? That way, even if the attribute value is empty, the quotation marks provide something to color and draw the user's attention. Oxygen's Compare Files tool includes quotation marks in its attribute value highlighting.

File: src/reporter/format-utils.xsl [hunk 2]
```diff
@@ -305,12 +347,13 @@
...
 </xsl:function>
 
-<!-- Compares $node with $node-to-compare-with and returns an HTML class accordingly: 'same' or 'diff'
+<!-- Compares $node with $node-to-compare-with and returns an HTML class accordingly: 'same', 'inner-diff' or 'diff'
   Set $expected to true if $node is in Expected Result. Set false if in Actual Result. -->
 <xsl:function name="test:comparison-html-class" as="xs:string">
   <xsl:param name="node" as="node()" />
   <xsl:param name="node-to-compare-with" as="node()?" />
   <xsl:param name="expected" as="xs:boolean" />
+  <xsl:param name="for-name" as="xs:boolean" />
```
Comment (galtm) [inline src/reporter/format-utils.xsl]: The name of the `for-name` parameter seemed a bit mysterious until I looked at how you were calling the function. Maybe a comment similar to the explanation of the `expected` parameter would be useful, such as, "Set $for-name to true only when comparing the names of elements, attributes, or processing instructions."

File: src/reporter/format-utils.xsl (hunk 1 above)
Comment (AirQuick) [reply src/reporter/format-utils.xsl]: Sounds reasonable.
Pushed the suggested change ([result](https://htmlpreview.github.io/?https://github.com/xspec/xspec/blob/292979b0c2f2b425c040655f45a83226571f51f7/test/end-to-end/cases/expected/stylesheet/xspec-serialize-result.html#ELEM-692)) and updated the initial comment (https://github.com/xspec/xspec/pull/704#issue-335147787).

File: src/reporter/format-utils.xsl (hunk 2 above)
Comment (AirQuick) [reply src/reporter/format-utils.xsl]: Yes, it wasn't a good name. Updated to a more descriptive name and added a comment. ([change](https://github.com/xspec/xspec/pull/704/commits/292979b0c2f2b425c040655f45a83226571f51f7#diff-0a14ba5b65a347807a3dacfbf7ba4a93R352-R401))

Comment (galtm) [reply src/reporter/format-utils.xsl]: Thanks! I like the new name. (It helps that "focusing" is not a language keyword the way the word "for" is.)

I saw your latest commit and I think this PR is ready to be merged, unless you are waiting for other reviewers.

**Comment by AirQuick** [comment]
Merged. Thanks @galtm for suggesting improvements.

**Pull request merged**